cypress.config.ts

```diff
@@ -42,24 +44,31 @@ export default defineConfig({
     numTestsKeptInMemory: 10,
     experimentalRunAllSpecs: true,
     experimentalMemoryManagement: true,
-    setupNodeEvents(on, config) {
-      const fetchConfigurationByFile = file => {
-        const pathOfConfigurationFile = `config/cypress.${file}.json`;
+    async setupNodeEvents(on, config) {
+      console.log("Cypress config", {baseUrl: config.baseUrl, testEnv: config.env.testEnv});
 
-        return (
-          file && fs.readJson(path.join(__dirname, "./cypress/", pathOfConfigurationFile))
-        );
-      };
+      // Print out cypress log to the command line so it is recorded by the CI system
+      installLogsPrinter(on);
 
-      require('cypress-terminal-report/src/installLogsPrinter')(on);
+      function getEnvConfig() {
+        const testEnv = config.env.testEnv;
+        if (testEnv) {
+          const pathOfConfigurationFile = `config/cypress.${testEnv}.json`;
+          return fs.readJson(path.join(__dirname, "./cypress/", pathOfConfigurationFile));
+        } else {
+          return {};
+        }
+      }
+      const envConfig = await getEnvConfig();
+      const combinedConfig = { ...config, ...envConfig };
 
-      const environment = config.env.testEnv || 'dev';
-      // First, read environments.json.
-      return fetchConfigurationByFile(environment)
-        .then(envConfig => {
-          return require('@cypress/code-coverage/task')(on, { ...config, ...envConfig });
-        });
+      // Save the code coverage information after each test.
+      // This also modifies the combinedConfig so the modified config needs to be returned
+      codeCoverageTask(on, combinedConfig);
+
+      return combinedConfig;
     },
+    // Try changing default baseUrl to see how that effects github action
```

Comment:
As a grammar nerd, I can't help pointing out that this should be "affects", not "effects". (Sorry, I can't help myself.)

Reply:
No worries. Thanks for finding that. 👍